src/main/java/com/ita/if103java/ims/dto/UserDto.java

```diff
@@ -0,0 +1,215 @@
+package com.ita.if103java.ims.dto;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.ita.if103java.ims.dto.transfer.ExistData;
+import com.ita.if103java.ims.dto.transfer.NewData;
+import com.ita.if103java.ims.dto.transfer.NewDataAdmin;
+import com.ita.if103java.ims.dto.transfer.NewDataWorker;
+import com.ita.if103java.ims.entity.Role;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.*;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+@Component
+public class UserDto implements Serializable {
+
+    @Null(groups = {NewDataAdmin.class},
+        message = "This field must be filled with the auto-generator during registration")
+    @NotNull(groups = {ExistData.class, NewDataWorker.class},
+        message = "This field must not be empty")
+    private Long id;
+
+    @Null(groups = {NewDataWorker.class},
+        message = "This field must be empty until confirmation via email and a real registration of Worker")
+    @NotBlank(groups = {NewDataAdmin.class, ExistData.class},
+        message = "Please, enter these data")
+    private String firstName;
+
+    @Null(groups = {NewDataWorker.class},
+        message = "This field must be empty until confirmation via email and a real registration of Worker")
+    @NotBlank(groups = {NewDataAdmin.class, ExistData.class},
+        message = "Please, enter these data")
+    private String lastName;
+
+    @NotBlank(groups = {NewData.class, ExistData.class},
+        message = "Please, enter an email")
+    @Email(groups = {NewData.class, ExistData.class},
+        message = "Please, enter a valid email")
+    private String email;
+
+    @Null(groups = {NewDataWorker.class},
+        message = "This field must be empty until confirmation via email and a real registration of Worker")
+    @NotBlank(groups = {NewDataAdmin.class, ExistData.class},
+        message = "Please, enter these data")
+    @Size(min = 8, max = 32)
+    private String password;
+
+    @NotNull(groups = {NewData.class, ExistData.class})
+    private Role role;
+
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
+    private LocalDateTime createdDate;
```

Comment:
LocalDateTime is not good to use here, because there is no information about time zone. Better to use ZoneDateTime.
Also need to update pattern above to include timezone.

This comment is valid to updatedDate field too